src/coreclr/vm/jitinterface.cpp

```diff
@@ -7890,6 +7890,25 @@ CorInfoInline CEEInfo::canInline (CORINFO_METHOD_HANDLE hCaller,
         }
     }
 
+    // If the root level caller and callee modules do not have the same runtime
+    // wrapped exception behavior, and the callee has EH, we cannot inline.
+    _ASSERTE(!pCallee->IsDynamicMethod());
+    {
+        COR_ILMETHOD_DECODER header(pCallee->GetILHeader(), pCallee->GetMDImport(), NULL);
+        if (header.EHCount() > 0)
+        {
+            Module* pCalleeModule = pCallee->GetModule();
+            Module* pRootModule = pOrigCaller->GetModule();
+
+            if (pRootModule->IsRuntimeWrapExceptions() != pCalleeModule->IsRuntimeWrapExceptions())
```

Comment:
These checks are catched cheap bit tests. We should do them first, and run the more expensive IL header decoder only when the modes do not match.

Reply:
Also looks like there are still issues with (lack of) an IL header.

Reply:
Likely I need to handle transient methods here too.